.github/workflows/target-branch.yml

```diff
@@ -0,0 +1,29 @@
+name: target-branch
+
+# Run this workflow *in the context of the main branch* when a pull request is
+# opened or reopened targeting the main branch.
+#
+# NOTE: Pull requests that change this file do not run those changes during the
+# "pull_request_target" event.
+on:
+  pull_request_target:
+    branches:
+      - main
```

Comment:
This differs slightly from the instructions, which say, "do not select `main` nor any other branch." I believe we can match that here [using `branches-ignore`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore):

```yaml
    branches-ignore:
      - integration
```

Reply:
Yeah, seems like this would be the way to go.

However, I have not found any reasonable way to test this yet. I tried a fork and it didn't work out well -- maybe I need to create an independent repo to trick it -- basically the problem is I can never get this workflow to run.

Reply:
The [`pull_request_target` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) is slightly different from the typical `pull_request` event. In general, it does things according to the _base_ or _target_ branch of the pull request, rather than the contents of the pull request itself.

The benefit is that it runs according to what is already committed, and there's no way for someone to submit a pull request that bypasses it. The drawback is that there's no way to trigger it until _after_ it is committed.

I'll make the change and figure out the behavior on my fork. I suspect that the workflow definition needs to be committed to the `main` branch before PRs to the `main` branch will be rejected.

Reply:
> I suspect that the workflow definition needs to be committed to the main branch before PRs to the main branch will be rejected.

That's pretty much what I found -- I think... It can always be patched if needed. It wouldn't be the first bad thing we committed to main.

But if you can get it as right looking as possible that would be helpful. Thanks!

Reply:
I tried some combinations in https://github.com/cbandy/pgbackrest/pull/2. The target branch (`main`) needs the workflow committed.